src/validation.cpp

```diff
@@ -5082,3 +5083,48 @@ void ChainstateManager::MaybeRebalanceCaches()
         }
     }
 }
+
+constexpr int SNAPSHOT_NAME_LEN = 75; // "chainstate_" + 64 hex characters for blockhash.
+
+static bool IsPathSnapshotDatadir(const fs::path& datadir_path)
+{
+    return fs::PathToString(datadir_path.filename()).length() == SNAPSHOT_NAME_LEN &&
+        fs::PathToString(datadir_path.filename()).substr(0,11) == "chainstate_";
+}
+
+static uint256 PathToSnapshotHash(const fs::path& datadir_path)
+{
+    assert(IsPathSnapshotDatadir(datadir_path));
+    return uint256S(fs::PathToString(datadir_path.filename()).substr(11));
+}
+
+static std::optional<fs::path> FindSnapshotChainstateDatadir()
+{
+    fs::path found;
+
+    for (fs::directory_iterator it(gArgs.GetDataDirNet()); it != fs::directory_iterator(); it++) {
+        if (fs::is_directory(*it) && !fs::is_empty(*it) && IsPathSnapshotDatadir(it->path()))
+        {
+            if (found.empty()) {
+                found = it->path();
+                LogPrintf("[snapshot] found snapshot datadir %s\n", fs::PathToString(found));
+            } else {
+                LogPrintf("[snapshot] WARNING - detected multiple snapshot " /* Continued */
```

Comment:
At least one improvement of the log message could be to advice the node operator on what they should do, e.g deleting the oldest snapshot one or whatever could be appropriate.